Makefile

```diff
@@ -54,12 +56,17 @@ $(tests_binary): $(SOURCE_FILES) | $(BUILD_DIR)
 	$(GET_DEPENDENCIES_WITH)
 	$(PONYC) -o $(BUILD_DIR) $(SRC_DIR)
 
-build-examples: $(EXAMPLES_BINARIES)
+build-examples: $(EXAMPLES_BINARIES) $(WEBSOCKET_EXAMPLES_BINARY)
 
 $(EXAMPLES_BINARIES): $(BUILD_DIR)/%: $(SOURCE_FILES) $(EXAMPLES_SOURCE_FILES) | $(BUILD_DIR)
 	$(GET_DEPENDENCIES_WITH)
 	$(PONYC) -o $(BUILD_DIR) $(EXAMPLES_DIR)/$*
 
+$(WEBSOCKET_EXAMPLES_BINARY): $(BUILD_DIR)/%: $(SOURCE_FILES) $(WEBSOCKET_EXAMPLE_SOURCE_FILES) | $(BUILD_DIR)
+	cd $(EXAMPLES_DIR)/$* && \
+		$(GET_DEPENDENCIES_WITH) && \
+		$(PONYC) -o ../../$(BUILD_DIR) $(EXAMPLES_DIR)/$*
+
```

Comment:
Why is this separate? having different commands to build examples seems like a footgun for people maintaining. (I'm think of a frustrated Sean in the future).

Reply:
I separated it out because it needs to fetch its own dependencies.